src/components/editor/MonacoEditor.tsx

```diff
@@ -117,12 +144,17 @@ function MonacoEditor({
                             alignItems: 'center',
                         }}
                     >
-                        {hasServerChanges ? (
+                        {/* TODO (editor) Need to get the save failure state working */}
+                        {status === EditorStatus.INVALID ||
+                        status === EditorStatus.SAVE_FAILED ? (
+                            <Invalid iconSize={ICON_SIZE} />
+                        ) : status === EditorStatus.OUT_OF_SYNC ? (
                             <ServerDiff
                                 iconSize={ICON_SIZE}
                                 onMerge={handlers.merge}
                             />
-                        ) : isChanging === null ? null : isChanging ? (
+                        ) : status === EditorStatus.IDLE ? null : status ===
+                          EditorStatus.EDITING ? (
                             <Saving iconSize={ICON_SIZE} />
                         ) : (
                             <Saved iconSize={ICON_SIZE} />
```

Comment:
Logic here:

1. Check if things are invalid or failed (failed will have it's own UX soon)
2. Check if there are server changes (not implemented until post MVP)
3. Are things idle -> show nothing
4. Are we saving -> Show saving
5. Default -> "Saved"